cocotb/scheduler.py

```diff
@@ -720,7 +720,10 @@ def schedule(self, coroutine, trigger=None):
             self.add(self._pending_coros.pop(0))
 
     def finish_test(self, exc):
```

Comment:
This can be moved to `RegressionManager` and `finish_scheduler` can be removed as redundant.

Reply:
I would also add a docstring explaining it is designed to be called outside of a scheduled test/task.

Reply:
I would like to move them, these functions don't really belong here anymore. @eric-wieser is that permissible? Will add a docstring and make them private methods as well.

Reply:
So I did the change.